### PR TITLE
Use protobuf IDL for echo service

### PR DESCRIPTION
**Pull request by tiziano88**
Also rename runtime to service, since it is not running anything else, and it is really just a service implementation for the interface defined in the IDL.

Note that this is a breaking change and will require changing the serialization on the untrusted launcher side too.

Ref #3385

### CODE REVIEW

**Review by conradgrobler — APPROVED** [review]

**Review by ipetr0v — APPROVED** [review]

File: testing/oak_echo_service/proto/oak_echo_service.proto [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+message EchoRequest {
+    bytes body = 1;
+}
+
+message EchoResponse {
+    bytes body = 1;
+}
+
+service EchoService {
+    // method_id: 1988
```
Comment (ipetr0v) [inline testing/oak_echo_service/proto/oak_echo_service.proto:30]: What is number?

File: testing/oak_echo_service/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Echo Service
+
+No-std compatible implementation of the business logic of the Echo Service.
+
+The interface of the service is defined using the Oak IDL in
+`proto/oak_echo_service.proto`, while the Rust code in this crate implements it.
```
Comment (ipetr0v) [inline testing/oak_echo_service/README.md:6]: Could you make it a link to the file?

File: oak_protobuf_idl_tests/oak.protobuf_idl.tests.rs.txt [hunk 3]
```diff
@@ -38,32 +37,29 @@ impl <S: TestService> ::oak_protobuf_idl::Transport for TestServiceServer<S> {
 
 impl <S: TestService> TestServiceServer<S> {
     fn invoke_inner(&mut self, request_bytes: &[u8]) -> Result<::prost::alloc::vec::Vec<u8>, ::oak_protobuf_idl::Status> {
-        use ::prost::Message;
         let request = ::oak_protobuf_idl::Request::decode(request_bytes).map_err(|err| {
             ::oak_protobuf_idl::Status::new_with_message(
                 ::oak_protobuf_idl::StatusCode::Internal,
-                format!("Client failed to deserialize the response: {:?}", err),
+                ::oak_protobuf_idl::format!("Client failed to deserialize the response: {:?}", err),
```
Comment (ipetr0v) [inline oak_protobuf_idl_tests/oak.protobuf_idl.tests.rs.txt:43]: Why is this a `.txt` file? And is there an instruction on how to update it?

File: oak_protobuf_idl_build/Cargo.toml [hunk 4]
```diff
@@ -6,5 +6,5 @@ license = "Apache-2.0"
 
 [dependencies]
 anyhow = "*"
-prost = "*"
+prost = { version = "*", default-features = false }
```
Comment (ipetr0v) [inline oak_protobuf_idl_build/Cargo.toml:9]: What's the difference between `oak_protobuf_idl` and `oak_protobuf_idl_build` ?

**Review by ipetr0v — APPROVED** [review]

File: testing/oak_echo_service/proto/oak_echo_service.proto (hunk 1 above)
Comment (tiziano88) [reply testing/oak_echo_service/proto/oak_echo_service.proto:30]: Just a random number for testing.

File: testing/oak_echo_service/README.md (hunk 2 above)
Comment (tiziano88) [reply testing/oak_echo_service/README.md]: Done.

File: oak_protobuf_idl_tests/oak.protobuf_idl.tests.rs.txt (hunk 3 above)
Comment (tiziano88) [reply oak_protobuf_idl_tests/oak.protobuf_idl.tests.rs.txt:43]: yes, it's currently explained in the generate.rs file already. But I can add a README.

File: oak_protobuf_idl_build/Cargo.toml (hunk 4 above)
Comment (tiziano88) [reply oak_protobuf_idl_build/Cargo.toml:9]: the _build version is used in build scripts. It's the same distinction between prost and prost-build, for instance.

File: testing/oak_echo_service/proto/oak_echo_service.proto (hunk 1 above)
Comment (ipetr0v) [reply testing/oak_echo_service/proto/oak_echo_service.proto:30]: Should it be 0, since previously we were using 0,1,2 for method IDs in flatbuffers?

**Pull request merged**